src/Segmentation/Watersheds/SegmentWithWatershedAndDistanceMap/Code.py

```diff
@@ -0,0 +1,151 @@
+#        Raising and lowering the Level influences the number of segments
+#        in the basic segmentation that are merged to produce the final output.
+#        A level of 1.0 is analogous to flooding the image up to a
+#        depth that is 100 percent of the maximum value in the image.
+#        A level of 0.0 produces the basic segmentation, which will typically be very oversegmented.
+#        Level values of interest are typically low (i.e. less than about 0.40 or 40%),
+#        since higher values quickly start to undersegment the image.
+
+
+import argparse
+
+import itk
+import numpy as np
+
+parser = argparse.ArgumentParser(
+    description="Segment an image using the watershed method and the signed Maurer distance map."
+)
+parser.add_argument("input_image_filename")
+parser.add_argument("reversed_input_output_filename")
+parser.add_argument("distance_map_output_filename")
+parser.add_argument("watershed_output_filename")
+parser.add_argument("segmentation_result_output_filename")
+parser.add_argument("binarizing_radius")
+parser.add_argument("majority_threshold")
+parser.add_argument("watershed_threshold")
+parser.add_argument("level")
+parser.add_argument("cleaning_structuring_element_radius")
+args = parser.parse_args()
+
+dimension = 3
+
+uchar_pixel_type = itk.UC
+input_image_type = itk.Image[uchar_pixel_type, dimension]
+
+float_pixel_type = itk.F
+float_image_type = itk.Image[float_pixel_type, dimension]
+
+rgb_pixel_type = itk.RGBPixel[uchar_pixel_type]
+RGBImageType = itk.Image[rgb_pixel_type, dimension]
+
+stack_image = itk.imread(args.input_image_filename)
+
+# Create bubble image: get a binarized version of the input image
+index_radius = itk.Size[dimension]()
+index_radius.Fill(args.binarizing_radius)
+
+voting_binary_hole_filling_image_filter = itk.VotingBinaryIterativeHoleFillingImageFilter(
+    stack_image,
+    radius=index_radius,
+    background_value=0,
+    foreground_value=255,
+    majority_threshold=args.majority_threshold,
+)
+
+voting_binary_hole_filling_image_filter.Update()
+
+bubble_image = voting_binary_hole_filling_image_filter.GetOutput()
+
+# Write bubble image
+itk.imwrite(bubble_image, args.reversed_input_output_filename)
+
+# Watershed on bubbles
+dimension = len(np.shape(bubble_image))
+
+# Convert to itk array and normalize
+multiply_image_filter = itk.MultiplyImageFilter[input_image_type, input_image_type, float_image_type]()
+multiply_image_filter.SetInput(bubble_image)
+multiply_image_filter.SetConstant(255.0)
+
+multiply_image_filter.Update()
+
+bubble_image_preclamp = multiply_image_filter.GetOutput()
+
+clamp_filter = itk.ClampImageFilter(
+    bubble_image_preclamp,
+    bounds=(0, 255),
+)
+
+bubble_image_clamp = clamp_filter.Update()
+
+itk_vol_img = clamp_filter.GetOutput()
+
+# Get the distance map of the input image
+distance_map_image_filter = itk.SignedMaurerDistanceMapImageFilter(itk_vol_img, inside_is_positive=False)
```

Comment:
`distance_image = itk.SignedMaurerDistanceMapImageFilter(itk_vol_img, inside_is_positive=False)` and
`distance_image = itk.signed_maurer_distance_map_image_filter(itk_vol_img, inside_is_positive=False)` should be equivalent, and both should be executions of the filter producing an image, not a filter instance. The second, snake case, is the preferred pythonic style.

Reply:
Thanks for the review @dzenanz. Done in 409b6ee. For now, I have not tried to run the code 😅 , and given that I do not use Python ITK I'm making such mistakes.